lib/active_record/connection_adapters/for_alter.rb

```diff
@@ -0,0 +1,91 @@
+require 'active_record/connection_adapters/mysql/schema_statements'
+
+module ForAlterStatements
+  class << self
+    def included(_)
+      STDERR.puts 'Including for_alter statements'
+    end
+  end
+
+  def bulk_change_table(table_name, operations) #:nodoc:
+    sqls = operations.flat_map do |command, args|
+      table = args.shift
+      arguments = args
+
+      method = :"#{command}_for_alter"
+
+      raise "Unknown method called : #{method}(#{arguments.inspect})" unless respond_to?(method, true)
+      send(method, table, *arguments)
```

Comment:
I'd rather keep the encapsulation using `public_send`.

Reply:
Makes sense! I've made that change